Sources/ComposableArchitecture/ViewStore.swift

```diff
@@ -69,21 +69,49 @@ public final class ViewStore<State, Action>: ObservableObject {
   ///     equal, repeat view computations are removed.
   public init(
     _ store: Store<State, Action>,
-    removeDuplicates isDuplicate: @escaping (State, State) -> Bool
+    removeDuplicates isDuplicate: @escaping (State, State) -> Bool,
+    instrumentation: Instrumentation = .shared
   ) {
-    self._send = { store.send($0) }
+    var actionString = ""
+    let eventInfo = Instrumentation.EventInfo(
+      type: String(describing: ViewStore.self),
+      action: actionString,
+      tags: [:]
+    )
+    self._send = {
+      actionString = debugCaseOutput($0)
+      var sendEventInfo = eventInfo
+      sendEventInfo.action = actionString
+      instrumentation.viewStore?.willSend(sendEventInfo)
```

Comment:
Wondering if there's a way we could keep the "event-building" in the `Instrumentation`, so the added code here becomes much smaller, and we could for example enable/disable instrumentation on the fly, and once it's disabled we wouldn't be paying for building the strings for actions etc.

```
let event = instrumentation.viewStore?.willSend(self, action)
defer { instrumentation.viewStore?.didSend(event) }
```

This might also allow us to keep a map of action to strings, so we'd only do this once per action.
(Speaking of which - have you had a chance to profile this? Does it show up as relevant in instruments / when using it in a bigger project?)